tools/lib/bootlog.py

```diff
@@ -36,9 +36,13 @@ def datetime(self) -> datetime.datetime:
     return timestamp_to_datetime(self._timestamp)
 
   def __eq__(self, b) -> bool:
+    if not isinstance(b, Bootlog):
+      return False
```

Comment:
```suggestion
  def __eq__(self, b: Bootlog) -> bool:
```

Reply:
I can't do that, seems like you can't typehint methods like `__x__`

Reply:
```
tools/lib/bootlog.py:38: error: Argument 1 of "__eq__" is incompatible with supertype "object"; supertype defines the argument type as "object"
tools/lib/bootlog.py:38: note: This violates the Liskov substitution principle
tools/lib/bootlog.py:38: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#incompatible-overrides
tools/lib/bootlog.py:38: note: It is recommended for "__eq__" to work with arbitrary objects, for example:
tools/lib/bootlog.py:38: note:     def __eq__(self, other: object) -> bool:
tools/lib/bootlog.py:38: note:         if not isinstance(other, Bootlog):
tools/lib/bootlog.py:38: note:             return NotImplemented
tools/lib/bootlog.py:38: note:         return <logic to compare two Bootlog instances>
```